pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentPreprocessingJob.java

```
@@ -0,0 +1,479 @@
+
+    // Optional push location and table parameters. If set, will use the table config and schema from the push hosts.
+    String pushHostsString = _properties.getProperty(JobConfigConstants.PUSH_TO_HOSTS);
+    String pushPortString = _properties.getProperty(JobConfigConstants.PUSH_TO_PORT);
+    if (pushHostsString != null && pushPortString != null) {
+      _pushLocations =
+          PushLocation.getPushLocations(StringUtils.split(pushHostsString, ','), Integer.parseInt(pushPortString));
+    } else {
+      _pushLocations = null;
+    }
+
+    _logger.info("*********************************************************************");
+    _logger.info("enable.partitioning: {}", _enablePartitioning);
+    _logger.info("enable.sorting: {}", _enableSorting);
+    _logger.info("enable.resizing: {}", _enableResizing);
+    _logger.info("path.to.input: {}", _inputSegmentDir);
+    _logger.info("path.to.output: {}", _preprocessedOutputDir);
+    _logger.info("path.to.deps.jar: {}", _pathToDependencyJar);
+    _logger.info("push.locations: {}", _pushLocations);
+    _logger.info("*********************************************************************");
+  }
+
+  public void run() throws Exception {
+    if (!_enablePartitioning && !_enableSorting && !_enableResizing) {
+      _logger.info("Pre-computing job is disabled.");
+      return;
+    } else {
+      _logger.info("Starting {}", getClass().getSimpleName());
+    }
+
+    _fileSystem = FileSystem.get(_configuration);
+    final List<String> inputDataPath = Arrays.asList(_inputSegmentDir.split("\\s*,\\s*"));
+    Path outputDir = new Path(_preprocessedOutputDir);
+
+    if (_fileSystem.exists(outputDir)) {
+      _logger.warn("Found the output folder {}, deleting it", _preprocessedOutputDir);
+      _fileSystem.delete(outputDir, true);
+    }
+    JobPreparationHelper.setDirPermission(_fileSystem, outputDir, _defaultPermissionsMask);
+
+    _tableConfig = getTableConfig();
```

this will fail if we don't have push location set, right? so we should fail in constructor instead of just warn

---

It will fail if there's no push location or there's no table config.